tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/RabbitMQ/RabbitMQIntegration.cs

```diff
@@ -192,24 +195,31 @@ internal static void SetDataStreamsCheckpointOnConsume(Tracer tracer, Span span,
                 }
             }
 
-            var scope = RabbitMQIntegration.CreateScope(Tracer.Instance, out RabbitMQTags tags, "basic.deliver", parentContext: propagatedContext, spanKind: SpanKinds.Consumer, queue: queue, exchange: exchange, routingKey: routingKey);
+            var scope = RabbitMQIntegration.CreateScope(Tracer.Instance, out var tags, "basic.deliver", parentContext: propagatedContext, spanKind: SpanKinds.Consumer, queue: queue, exchange: exchange, routingKey: routingKey);
             if (tags != null)
             {
-                tags.MessageSize = body?.Length.ToString() ?? "0";
+                if (body.Instance is not null)
+                {
+                    tags.MessageSize = body.Length.ToString() ?? "0";
+                }
+
+                if (scope is not null)
```

Comment:
very nit but as there's no guarantee scope is null if tags are null (even though we know it's the case), I would have left the check for scope nullity outside of this first if.

Reply:
_sigh_ yeah. 


The fact it's not possible to mark `tags` as non-null when scope is non-null is really annoying 🙁  We could "fix" it by changing the signature of `CreateScope` to 

```csharp
bool CreateScope([NotNullWhen(true)] out Span? span, [NotNullWhen(true)] out RabbitMqTags? tags)
```

But that's a bit messy too 🙄

Reply:
> The fact it's not possible to mark tags as non-null when scope is non-null is really annoying

Agreed. I thought of the attributes too. For me it's one or the other but not the current implem. That said, it's nitpicky as the chance to have one null in the future is clearly almost nil

Reply:
Yeah. I moved the null to the outer check for simplicity, and we can rethink it later